// await 
{
	synchronized (x10.lang.Runtime.here()) {
  		try {
  			while (!(#0)) {
  				x10.lang.Runtime.here().wait();    			
    		}
    		x10.lang.Runtime.here().notifyAll();
  		} catch (Throwable e) {
  			e.printStackTrace();
  			x10.lang.Runtime.runtime.java.log(java.util.logging.Level.SEVERE, "Unexpected exception during conditional atomic.", e);
  			throw new RuntimeException(e);
  		}
	}
}